app/(auth)/auth/callback/route.ts

```diff
@@ -72,23 +73,37 @@ async function initializeUserIfNeeded(user: User) {
 	}
 }
 
+interface ProviderConfig {
+	name: OAuthProvider;
+	isMatch: (session: Session) => boolean;
+}
+
+const oauth_providers: ProviderConfig[] = [
+	{
+		name: "github",
+		isMatch: (session) =>
+			session.user.app_metadata.providers.includes("github"),
+	},
+	{
+		name: "google",
+		isMatch: (session) =>
+			session.user.app_metadata.providers.includes("google"),
+	},
+] as const;
+
 // store accessToken and refreshToken
 async function storeProviderTokens(user: User, session: Session) {
 	const { provider_token, provider_refresh_token } = session;
 	if (!provider_token) {
 		throw new Error("No provider token found");
 	}
 
-	let provider = "";
-	// https://docs.github.com/ja/authentication/keeping-your-account-and-data-secure/about-authentication-to-github#github-%E3%81%AE%E3%83%88%E3%83%BC%E3%82%AF%E3%83%B3%E3%83%95%E3%82%A9%E3%83%BC%E3%83%9E%E3%83%83%E3%83%88
-	if (provider_token.startsWith("ghu_")) {
-		provider = "github";
-	}
-	// TODO: add another logic for other providers
-	if (provider === "") {
-		throw new Error("No provider found");
+	const matchedProvider = oauth_providers.find((p) => p.isMatch(session));
```

Comment:
In my Supabase user object, `app_metadata` is as follows:

```
app_metadata: { provider: 'email', providers: [ 'email', 'github' ] },
```

After I connecting my Google account, it may look like this:

```
app_metadata: { provider: 'email', providers: [ 'email', 'github', 'google' ] },
```

I am not sure how the order of app_metadata.providers is determined. 
However this logic will attempt to use GitHub provider's access token and refresh token, even if I've just authenticated with Google OAuth.

Reply:
I [reverted it](https://github.com/giselles-ai/giselle/pull/107/commits/23c6afe13bf20ead9577386667c93060fd19226f) and implemented "settings/" page.

would you re-review the codes? @satococoa